src/System.Management.Automation/engine/Utils.cs

```diff
@@ -202,7 +202,7 @@ internal static BigInteger ParseBinary(ReadOnlySpan<char> digits, bool unsigned)
             // Calculate number of 8-bit bytes needed to hold the input,  rounded up to next whole number.
             // This value will also be used as the indexer for our array.
             int outputByteWalker = (digits.Length + 7) / 8;
-            Span<byte> outputBytes = new byte[outputByteWalker--];
+            Span<byte> outputBytes = outputByteWalker <= 512 ? stackalloc byte[outputByteWalker--] : new byte[outputByteWalker--];
```

Comment:
Please don't use `outputByteWalker--` here - very bad readability is way to bugs.
I'd add new local variable - one for size/length, second for last index if needed.

Reply:
I'll take care of this momentarily. 😄